### PR TITLE
docs: Update README and bump version to 1.4.2

### PR DESCRIPTION
**Pull request by shenxianpeng**
I assume to bump the version to 1.4.2 because

1. The current version in setup.py is incorrect, it is 1.3.1
2. update README based on suggestion in https://github.com/shenxianpeng/cpp-linter-action/issues/55#issuecomment-1106666257

### CODE REVIEW

**Comment by shenxianpeng** [comment]
Tested looks good 
```bash
python3 -m pip install git+https://github.com/shenxianpeng/cpp-linter-action@v1.4.2

gitpod /workspace/cpp-linter-action (update-readme) $ pip3 show cpp-linter
Name: cpp-linter
Version: 1.4.2
Summary: Bootstrapper for docker's ENTRYPOINT executable.
Home-page: https://github.com/2bndy5/cpp-linter-action
Author: Brendan Doherty
Author-email: 2bndy5@gmail.com
License: MIT
Location: /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages
Requires: pyyaml, requests
Required-by: 
```

**Comment by 2bndy5** [comment]
I'm curious if using `@v1` will allow us to only require future changes when bumping the major version.

[According to the github docs](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-release-management-for-actions), using `v1` can be shorthand for `v1.x.x`. I think this means we would have to keep moving a separate tag named `v1` to the latest released commit (which is currently tagged with `v1.4.2`). This would be easier than updating the README for every release.

**Comment by 2bndy5** [comment]
> I think this means we would have to keep moving a separate tag named v1 to the latest released commit

Or we could create a branch named `v1` and keep pushing deployment-ready commits to that branch (likely cherry-picked from master). But this seem like a bit more work than moving a tag around.

**Comment by 2bndy5** [comment]
> The current version in setup.py is incorrect, it is 1.3.1

Ideally we wouldn't need to keep updating this in setup.py if we use the `setuptools_scm` lib as a build dependency. However, I neglected to do this since it adds to the action setup time ([`setuptools_scm`](https://pypi.org/project/setuptools-scm/) is only available via pip).

**Comment by shenxianpeng** [comment]
From the tests you run, `v1` makes a lot of sense, and it also works with `python3 -m pip install git+https://github.com/shenxianpeng/cpp-linter-action@v1`

If every release needs to build and upload a wheel to be published to pypi seems to add a bit more work.

File: setup.py [hunk 1]
```diff
@@ -17,14 +15,17 @@
     # setup_requires=["setuptools_scm"],
     version="1.4.2",
     description=__doc__,
-    long_description=".. warning:: this is not meant for PyPi (yet)",
-    author="Brendan Doherty",
+    long_description=(
+        "A python package that powers the github action named cpp-linter-action. "
+        + f"See `the github repository README <{REPO}#readme>`_ for full details."
+    ),
+    author="Brendan Doherty, Peter Shen",
```
Comment (shenxianpeng) [inline setup.py:22]:
```suggestion
    author="Brendan Doherty",
```

**Review by shenxianpeng — COMMENTED** [review]
LGTM

**Comment by 2bndy5** [comment]
>If every release needs to build and upload a wheel to be published to pypi seems to add a bit more work.

I can easily automate this with CD workflow, but I'm not sure it is worth it right now. Maybe in the future when we have patched in support for other CI services...

File: setup.py (hunk 1 above)
Comment (2bndy5) [reply setup.py:22]: You don't want your name on this?! I didn't add you email to `author_email` field.

**Comment by shenxianpeng** [comment]
To save setup time or add a CI workflow for this? Yes, maybe in the future ...

**Review by 2bndy5 — APPROVED** [review]
Since I pushed the `v1` tag (and tested it on the test repo with success), I pushed a commit that updated the README to use `v1` instead of the full tag `v1.4.2`.

**Comment by 2bndy5** [comment]
> To save setup time

Recent tests (on the test repo) show that pip installing from the repo directly (instead of pypi) only takes about 12 seconds. So, I'm not sure how much time it would actually save (I'm guessing around 5 seconds).

File: setup.py (hunk 1 above)
Comment (shenxianpeng) [reply setup.py:22]: You are the author of the Python code, I may be a future contributor, shame, but thank you!

**Comment by 2bndy5** [comment]
If you're ready to merge, I can move around the tags afterward (and remove the "pre-release" flag).

**Comment by shenxianpeng** [comment]
5 seconds sounds good! OK, I. am going to merge

**Comment by 2bndy5** [comment]
FYI, I had to
1. delete the tags
2. push the changes
3. re-create the tags
4. push the changes again

I think this is why the pre-release description of v1.4.2 got removed. BTW, having the tags on a branch that was not the master branch proved to be an additional obstacle.

**Comment by shenxianpeng** [comment]
I did

* deleted the pre-release 
* recreate a release v1.4.2 using the tag you created.

I added v1.4.2 to the update-readme branch for testing, it really cause some inconvenience.

File: setup.py (hunk 1 above)
Comment (2bndy5) [reply setup.py:22]: I consider feedback, review, and testing as contributions.. You deserve it!

**Comment by 2bndy5** [comment]
oh! In the future, I'll leave the releases in your capable hands.

**Comment by shenxianpeng** [comment]
No problem, I am a release engineer in my other capacity 😆

**Pull request merged**